### ISSUE TITLE
Softplus differs depending on number of elements in tensor

### ISSUE THREAD

**Issue by mboratko** [pytorch/pytorch#48235]
## 🐛 Bug

The output of `F.softplus` is different depending on the number of elements in the input tensor.

## To Reproduce
```
import torch
import torch.nn.functional as F
problem_value = -1.1052913665771484
A = torch.zeros(15)
A[0] = problem_value
assert F.softplus(A)[0] == F.softplus(A[0])
A = torch.zeros(16)
A[0] = problem_value
assert F.softplus(A)[0] != F.softplus(A[0])
```

## Expected behavior

The value of softplus should not depend on the number of elements in the tensor.
In particular, the last line in the code sample above should throw an error.

## Environment

PyTorch version: 1.7.0
Is debug build: True
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Mac OSX 10.15.7 (x86_64)
GCC version: Could not collect
Clang version: 12.0.0 (clang-1200.0.32.21)
CMake version: Could not collect

Python version: 3.8 (64-bit runtime)
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.19.2
[pip3] torch==1.7.0
[conda] blas                      1.0                         mkl
[conda] mkl                       2019.4                      233
[conda] mkl-service               2.3.0            py38hfbe908c_0
[conda] mkl_fft                   1.2.0            py38hc64f4ea_0
[conda] mkl_random                1.1.1            py38h959d312_0
[conda] numpy                     1.19.2           py38h456fd55_0
[conda] numpy-base                1.19.2           py38hcfb5961_0
[conda] pytorch                   1.7.0                   py3.8_0    pytorch

## Additional context

This was originally observed where the split between the number of elements in the tensor was much higher, say ~ 1000 and 20,000. It seems to depend both on the particular value (`problem_value` in my example) and the number of elements.

**Comment by izdeby** [comment]
Marking hi pri due to correctness issue.

**Comment by mruberry** [comment]
This is probably expected.

How different are the values you're seeing?

PyTorch attempts to maximize performance by using different algorithms or computation techniques depending on the structure of inputs it's given. Floating point math is imperfect, so these differences can have different numeric results. In this case the assertion is testing for bitwise accuracy, which is not guaranteed.

I would expect, however, the actual difference in these values to be very small.

**Comment by ngimel** [comment]
The difference on my system is 3e-8 which is acceptable for vectorized vs non-vectorized implementation, removing hi-pri.
